### PR TITLE
Add instructions to update documentation

### PR DESCRIPTION
**Pull request by SylvainJuge**
Instructions on how to update and build documentation were missing into `CONTRIBUTING.MD`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -223,6 +223,16 @@ See [`apm-agent-core/README.md`](apm-agent-core/README.md)
 
 See [`apm-agent-plugins/README.md`](apm-agent-plugins/README.md)
 
+### Documenting
+
+Documentation is generated text files stored in `docs` folder using [AsciiDoc](http://asciidoc.org/) format.
```
Comment (felixbarny) [inline CONTRIBUTING.md]: Only the configuration documentation (`configuration.asciidoc`) is generated, the rest is manually written

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -223,6 +223,16 @@ See [`apm-agent-core/README.md`](apm-agent-core/README.md)
...
+The ``configuration.asciidoc`` file is generated from running `co.elastic.apm.agent.configuration.ConfigurationExporterTest`.
+
+A preview of documentation is generated for each pull-request.
```
Comment (felixbarny) [inline CONTRIBUTING.md]:
```suggestion
A preview of the documentation is generated for each pull-request.
Click on the build `Details` of the `elasticsearch-ci/docs` job and go to the bottom of the `Console Output` to see the link.
```

**Comment by SylvainJuge** [comment]
Jenkins run the tests please

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -223,6 +223,18 @@ See [`apm-agent-core/README.md`](apm-agent-core/README.md)
 
 See [`apm-agent-plugins/README.md`](apm-agent-plugins/README.md)
 
+### Documenting
+
+HTML Documentation is generated from text files stored in `docs` folder using [AsciiDoc](http://asciidoc.org/) format.
```
Comment (felixbarny) [inline CONTRIBUTING.md]:
```suggestion
[HTML Documentation](https://www.elastic.co/guide/en/apm/agent/java/current/index.html) is generated from text files stored in `docs` folder using [AsciiDoc](http://asciidoc.org/) format.
```

Comment (bmorelli25) [reply CONTRIBUTING.md]: We actually use [Asciidoctor](https://asciidoctor.org/) now, not Asciidoc.

**Pull request merged**